contentcuration/contentcuration/viewsets/file.py

```diff
@@ -29,6 +31,39 @@
 from contentcuration.viewsets.sync.constants import CONTENTNODE
 from contentcuration.viewsets.sync.utils import generate_update_event
 
+class StrictFloatField(serializers.FloatField):
+    def to_internal_value(self, data):
+        # If data is a string, reject it even if it represents a number.
+        if isinstance(data, str):
+            raise serializers.ValidationError("A valid number is required.")
+        return super().to_internal_value(data)
+
+# New Serializer for validating upload_url inputs
+class FileUploadURLSerializer(serializers.Serializer):
+    """
+    Serializer to validate inputs for the upload_url endpoint.
+    Required:
+      - size: a float value
+      - checksum: a 32-digit hex string
+      - filename: a string (note: mapped from request.data['name'])
+      - file_format: a valid file format choice from file_formats.choices
+      - preset: a valid preset choice from format_presets.choices
+    Optional:
+      - duration: a number that will be floored to an integer and must be > 0
+    """
+    size = serializers.FloatField(required=True)
+    checksum = serializers.RegexField(regex=r'^[0-9a-fA-F]{32}$', required=True)
```

Comment:
OK to just enforce lower case here - we don't have a use case for upper case letters in the hex string.